packages/core/src/messagingProtocols/proteus/Utility/SessionHandler/SessionHandler.ts

```diff
@@ -0,0 +1,258 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {PreKey} from '@wireapp/api-client/lib/auth';
+import {QualifiedUserClients, UserClients} from '@wireapp/api-client/lib/conversation';
+import {QualifiedId, UserPreKeyBundleMap} from '@wireapp/api-client/lib/user';
+import {CoreCrypto} from '@wireapp/core-crypto/platforms/web/corecrypto';
+import {Decoder} from 'bazinga64';
+import {Logger} from 'logdown';
+
+import {APIClient} from '@wireapp/api-client';
+
+import {SessionId} from './SessionHandler.types';
+
+import {isQualifiedUserClients, isUserClients} from '../../../../util';
+import {preKeyBundleToUserClients} from '../PreKeyBundle';
+
+interface ConstructSessionIdParams {
+  userId: string | QualifiedId;
+  clientId: string;
+  useQualifiedIds?: boolean;
```

Comment:
Let's make this one mandatory to avoid mis-generated sessionIds (and since this one is only supposed to be used internally)